apps/member-profile/app/routes/_profile.resources.$id_.edit.tsx

```diff
@@ -93,6 +97,20 @@ export async function action({ params, request }: ActionFunctionArgs) {
     return json({ errors }, { status: 400 });
   }
 
+  // Editing resource with a new link that already exisits
+  if (data.link) {
+    const existingResource = await findResourceByUrl(data.link);
+
+    if (existingResource && existingResource.id !== params.id) {
+      const errors: Record<'message' | 'resourceId', string> = {
+        message: 'A resource with this link has already been added.',
+        resourceId: existingResource.id,
+      };
+
+      return json({ errors });
+    }
+  }
```

Comment:
Thoughts on moving some of this core logic inside of the `addResource` and `updateResource` functions?

If you're up to it, let's convert those functions to return a `Result` (see the `Result`, `fail`, and `success` utilities). You can see an example of a function that returns a `Result` type [here](https://github.com/colorstackorg/oyster/blob/e727c735a9f552412929464607715c5f0ad9c5e0/packages/core/src/infrastructure/ai.ts#L71). Feel free to adapt the `Result` in the failure case to include a `data` field so that you can grab that `resourceId` as well.

Reply:
Thank you Chika! I went ahead and made an update to the core file with the [`Result`](https://github.com/colorstackorg/oyster/pull/711/files#diff-d1dc3c60a9b08057b9aa3a31f4f6a140a9af1a96e063c5b67ee9d1d58dd22c24L3)type , so now we allow a `data` parameter to be passed into the "failure" case. I think this will be useful for other use cases as well which is why I did that for the underlying type. That simplified things a bit so that now we don't have to do a manual check when consuming the result of the `addResource` or `updateResource` functions, ie:

```
'duplicateResourceId' in `errors`
```